frontend/micro-ui/web/micro-ui-internals/packages/modules/templates/ApplicationDetails/Modal/AttendanceActionModal.js

```diff
@@ -0,0 +1,205 @@
+import { Loader, Modal, FormComposer } from "@egovernments/digit-ui-react-components";
+import React, { useState, useEffect } from "react";
+import { configAttendanceApproveModal, configAttendanceRejectModal, configAttendanceCheckModal } from "../config";
+
+const Heading = (props) => {
+  return <h1 className={props.className ? `heading-m ${props.className}` : "heading-m"}>{props.label}</h1>;
+};
+
+const Close = () => (
+  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="#FFFFFF">
+    <path d="M0 0h24v24H0V0z" fill="none" />
+    <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z" />
+  </svg>
+);
+
+const CloseBtn = (props) => {
+  return (
+    <div className="icon-bg-secondary" onClick={props.onClick}>
+      <Close />
+    </div>
+  );
+};
+
+const AttendanceActionModal = ({ t, action, tenantId, state, id, closeModal, submitAction, actionData, applicationData, businessService, moduleCode,applicationDetails,workflowDetails }) => {
+
+  let { loiNumber, estimateNumber } = Digit.Hooks.useQueryParams();
+   const [config, setConfig] = useState({});
+   const [defaultValues, setDefaultValues] = useState({});
+   const [approvers, setApprovers] = useState([]);
+   const [selectedApprover, setSelectedApprover] = useState({});
+   
+   const [department, setDepartment] = useState([]);
+   const [selectedDept,setSelectedDept] = useState({})
+
+   const [rejectionReason, setRejectionReason] = useState([]);
+   const [selectedReason,setSelectedReason] = useState([])
+
+   const [designation, setDesignation] = useState([]);
+   const [selectedDesignation,setSelectedDesignation] = useState({})
+
+  const rejectReasons = [
+    {
+      name: "Estimate Details are incorrect"
+    },
+    {
+      name: "Financial Details are incorrect"
+    },
+    {
+      name: "Agreement Details are incorrect"
+    },
+    {
+      name: "Vendor Details are incorrect"
+    },
+    {
+      name: "Attachments provided are wrong"
+    },
+    {
+      name: "Others"
+    },
+  ]
+
+  const { isLoading: mdmsLoading, data: mdmsData,isSuccess:mdmsSuccess } = Digit.Hooks.useCustomMDMS(
+    Digit.ULBService.getCurrentTenantId(),
+    "common-masters",
+    [
+      {
+        "name": "Designation"
+      },
+      {
+        "name": "Department"
+      }
+    ]
+  );
+
+  mdmsData?.["common-masters"]?.Designation?.map(designation => {
```

Comment:
for formatting any hook data send a function as default which will give u formatted response
ref : frontend/micro-ui/web/micro-ui-internals/packages/libraries/src/hooks/useCustomMDMS.js

Reply:
done